web-components/src/components/form/CreditRetireForm.tsx

```diff
@@ -0,0 +1,269 @@
+  availableTradableAmount: number;
+  onClose: () => void;
+}
+
+export interface FormValues {
+  retiredAmount: number;
+  note: string;
+  country: string;
+  stateCountry?: string;
+  postalCode?: string;
+}
+
+interface CreditRetireFieldsProps {
+  country: string;
+  batchDenom: string;
+  availableTradableAmount: number;
+}
+
+export const CreditRetireFields = ({
+  country,
+  batchDenom,
+  availableTradableAmount,
+}: CreditRetireFieldsProps): JSX.Element => {
+  const styles = useStyles();
+
+  const countryHandler = (countryCode: string): any => {
+    console.log('*** countryHandler', countryCode);
+  };
+
+  const stateCountryHandler = (stateCountry: string): any => {
+    console.log('*** stateCountryHandler', stateCountry);
+  };
+
+  return (
+    <>
+      <AmountField
+        name={'retiredAmount'}
+        label={'Amount to retire'}
+        availableAmount={availableTradableAmount}
+        batchDenom={batchDenom}
+        className={styles.textField}
+      />
+      <Title className={styles.groupTitle} variant="h5">
+        Transaction note
+      </Title>
+      <Field
+        name="note"
+        type="text"
+        label="Add retirement transaction details (stored in the transaction note)"
+        component={TextField}
+        className={styles.noteTextField}
+        optional
+      />
+      <Title className={styles.groupTitle} variant="h5">
+        Location of retirement
+      </Title>
+      <Description className={styles.description}>
+        Please enter a location for the retirement of these credits. This
+        prevents double counting of credits in different locations.
+      </Description>
+      <Grid container alignItems="center" className={styles.stateCountryGrid}>
+        <Grid item xs={12} sm={6} className={styles.stateCountryTextField}>
+          <LocationStateField
+            country={country}
+            triggerOnChange={stateCountryHandler}
+          />
+        </Grid>
+        <Grid item xs={12} sm={6} className={styles.stateCountryTextField}>
+          <LocationCountryField triggerOnChange={countryHandler} />
+        </Grid>
+      </Grid>
+      <Field
+        className={styles.postalCodeField}
+        component={ControlledTextField}
+        label="Postal Code"
+        name="postalCode"
+        optional
+      />
+    </>
+  );
+};
+
+export const validateCreditRetire = (
+  availableTradableAmount: number,
+  values: FormValues,
+  errors: FormikErrors<FormValues>,
+): FormikErrors<FormValues> => {
+  if (!values.country) {
+    errors.country = requiredMessage;
+  }
+
+  if (!values.retiredAmount) {
+    errors.retiredAmount = requiredMessage;
+  } else if (Math.sign(values.retiredAmount) !== 1) {
+    errors.retiredAmount = invalidAmount;
+  } else if (values.retiredAmount > availableTradableAmount) {
+    errors.retiredAmount = insufficientCredits;
+  }
+
+  return errors;
+};
+
+export const initialValues = {
+  retiredAmount: 0,
+  note: '',
+  country: 'US',
```

Comment:
why choosing this as initial value? I believe we should just leave it empty

Reply:
i agree, done!